lib/internal/Magento/Framework/Mail/Transport.php

```diff
@@ -18,7 +18,7 @@ class Transport extends \Zend_Mail_Transport_Sendmail implements \Magento\Framew
      * @param null $parameters
      * @throws \InvalidArgumentException
      */
-    public function __construct(\Magento\Framework\Mail\MessageInterface $message, $parameters = null)
+    public function __construct(\Magento\Framework\Mail\ExtendedMessageInterface $message, $parameters = null)
```

Comment:
Modification of constructor is not allowed as this is backward incompatible change. Please see http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/. Change of argument type to more strict is equal to removing all argument and introduction of new one.